File: clases 25-11-2025/Tipos de clases GO/clase_normal_con_interfaz/Libro.go
package main

import "fmt"

type ILibro interface {
	MostrarLibro()
}

type Libro struct {
	titulo string
	autor  string
}

func NewLibro(titulo string) ILibro {
	return Libro{titulo: titulo, autor: "Desconocido"}
}

func (l Libro) MostrarLibro() {
	fmt.Printf("Título: %s, Autor: %s\n", l.titulo, l.autor)
}

File: clases 25-11-2025/Tipos de clases c#/clase normal con interfaz/Libro.cs
public class Libro : ILibro
{
    public string Titulo { get; set; }
    public string Autor { get; set; } = "Desconocido";

    public Libro(string titulo)
    {
        Titulo = titulo;
    }

    public void MostrarLibro()
    {
        Console.WriteLine($"Título: {Titulo}, Autor: {Autor}");
    }
}